pkg/util/cache/basic_cache.go

```diff
@@ -10,12 +11,22 @@ type BasicCache struct {
 	m        sync.RWMutex
 	cache    map[string]interface{}
 	modified int64
+	eqFunc   func(interface{}, interface{}) bool
 }
 
 // NewBasicCache Creates new BasicCache
 func NewBasicCache() *BasicCache {
 	return &BasicCache{
-		cache: make(map[string]interface{}),
+		cache:  make(map[string]interface{}),
+		eqFunc: func(a, b interface{}) bool { return reflect.DeepEqual(a, b) },
```

Comment:
Note: You can also use `return NewBasicCacheWithEqualityFunc(func(a, b interface{}) bool { return reflect.DeepEqual(a, b) })`